### DISCUSSION TITLE
How do I dynamically set private attribute on instance

### DISCUSSION THREAD

**Discussion by MarcSkovMadsen** [pydantic/pydantic#3512] [Question]
Hi

I'm trying to convert Pydantic model instances to HoloViz Param instances.

I would like to store the resulting `Param` instance in a private attribute on the Pydantic instance. I can do this use `__setattr__` but then the private variable shows up in the `.dict()`. **How do I avoid this**?

## Example

```python
from datetime import datetime
from typing import List

from pydantic import BaseModel

def to_parameterized(model: BaseModel):
    parameterized = "A Parameterized Instance"
    object.__setattr__(model, "__parameterized__", parameterized)
    return parameterized

class User(BaseModel):
    id: int=123

user = User()
param_user = to_parameterized(user)
assert user.dict()=={"id": 123}
```

I would like this to run without error. But instead I get 

```bash
$ python 'script3.py'
Traceback (most recent call last):
  File "script3.py", line 16, in <module>
    assert user.dict()=={"id": 123}
AssertionError
```

**Comment by ryanbbernstein** [comment]
If you just want to be able to modify it while keeping it private, you can use `PrivateAttr`

For validation I found a workaround that allows modifying or reading from private attributes during construction.
```
from pydantic import BaseModel, PrivateAttr

class Model(BaseModel):
    public: str
    _private: str = PrivateAttr

    def _init_private_attributes(self) -> None:
        super()._init_private_attributes()
        self._private = "this works"
        # or
        if self._private:
            self.public = "new value"
````

This works for construction only and feels a bit hacky though.
